### PR TITLE
Reaches the point of sending a room swap request

### PR DESCRIPTION
**Pull request by Index01**
- creates rooms staff and guests
    - available for swap request button
    - must send jwt at top level request for auth module

### CODE REVIEW

File: backend/createStaffAndRooms.py [hunk 1]
```diff
@@ -78,7 +78,10 @@ def create_rooms_main(rooms_file, is_hardrock=False, force_roombaht=False):
                         number=elem['Room']
                         )
 
-            features = elem['Room Features (Accessibility, Lakeview, Smoking)'].lower()
+            try:
```
Comment (otakup0pe) [inline backend/createStaffAndRooms.py:81]: Are you still having parsing errors here? I'm not super comfortable with making this bit optional (unless we make it mutable) especially given #66

File: backend/reservations/views/rooms.py [hunk 2]
```diff
@@ -77,6 +78,7 @@ def room_list(request):
             elif(len(room['number'])==4):
                 room["floorplans"]=FLOORPLANS[int(room["number"][:2])]
             for room_type in room_types:
+                print(f'roomstype: {room_type}, room name t3; {room["name_take3"]}')
```
Comment (Index01) [inline backend/reservations/views/rooms.py:81]: sheisse

File: backend/reservations/auth.py [hunk 3]
```diff
@@ -13,29 +13,29 @@
 def authenticate(request):
     data = request.data
     try:
-        jwt_data=data["jwt"]
+        jwt_data=data["guest"]["jwt"]
     except KeyError:
-        logger.warning("JWT information missing ip:%s", request.META['REMOTE_ADDR'])
+        logger.warning("[-] JWT information missing ip:%s", request.META['REMOTE_ADDR'])
```
Comment (otakup0pe) [inline backend/reservations/auth.py:18]: What is the meaning behind the `[-]` / `[+]` prefixes?

File: backend/reservations/views/rooms.py (hunk 2 above)
Comment (otakup0pe) [inline backend/reservations/views/rooms.py:81]: `logger.debug` plz

**Review by otakup0pe — APPROVED** [review]
:ok: to go but would ❤️ it if we didn't have `print` in there.

File: backend/createStaffAndRooms.py (hunk 1 above)
Comment (Index01) [reply backend/createStaffAndRooms.py:81]: commented on the issue. 
Are we thinking we need these fields to compose the complete hotel name? 
it feels extra to me. would be nice to a have an extensible dataset but ultimately room number is wat the hotel mostly cares about ja?

Comment (Index01) [reply backend/createStaffAndRooms.py:81]: Also if is pref, i can use the new room csv. 
The one i have just contains the rooms types i wanted for 1to1 matching with the short_guest lists i made. i cans change tho!

File: backend/reservations/auth.py (hunk 3 above)
Comment (Index01) [reply backend/reservations/auth.py:18]: easy for me to spot in logs :rofl:

File: backend/reservations/views/rooms.py (hunk 2 above)
Comment (Index01) [reply backend/reservations/views/rooms.py:81]: deff

File: backend/createStaffAndRooms.py (hunk 1 above)
Comment (Index01) [reply backend/createStaffAndRooms.py:81]: Sheisse. forgot to do something about this. will lewk tmw

Comment (otakup0pe) [reply backend/createStaffAndRooms.py:81]: New room CSV (and any test fixtures we come up with) should use the format listed here. Note some of the parsing semantics will change once #59 lands.

File: backend/reservations/views/rooms.py (hunk 2 above)
Comment (otakup0pe) [reply backend/reservations/views/rooms.py]: `logger.debug` instead of `print` plz

File: backend/reservations/auth.py [hunk 4]
```diff
@@ -14,28 +14,28 @@ def authenticate(request):
     data = request.data
     try:
         jwt_data=data["jwt"]
-    except KeyError:
-        logger.warning("JWT information missing ip:%s", request.META['REMOTE_ADDR'])
+    except KeyError as e:
+        logger.warning("[-] JWT information missing ip:%s data: %s, e: %s", request.META['REMOTE_ADDR'], data, e)
```
Comment (otakup0pe) [inline backend/reservations/auth.py:18]: I could have swore I made this comment already but that was before `N` shots of fernet....

What is the meaning of the `[-]` prefix? Is it just to stand out?

File: backend/reservations/views/rooms.py [hunk 5]
```diff
@@ -63,7 +63,8 @@ def room_list(request):
                     .exclude(guest=None)
         guest_entries = Guest.objects.filter(email=email)
         try:
-            room_types = [Room.objects.filter(number=guest.room_number)[0].name_take3 for guest in guest_entries]
+            rt = [guest.room_number for guest in guest_entries if guest.room_number is not None]
```
Comment (otakup0pe) [inline backend/reservations/views/rooms.py:66]: Tite. This is to solve the admins not seeing their rooms? I wonder if mid term we could just add rooms to the base admin record and long term it's #61

**Review by otakup0pe — APPROVED** [review]
I'm assuming the auth refactor I made resulted in all the regressions because the data obj was different for different request. TY for catching another one, I feel better about where it is now 💖

File: backend/reservations/views/rooms.py (hunk 5 above)
Comment (Index01) [reply backend/reservations/views/rooms.py:66]: Ja was resulting in admins not able send swap reqs. Because room was None so room type had no match

**Pull request merged**